File: python/pc.py
from processor import processor
from disk import disk
from ram import ram

class pc(processor, disk, ram):
    processor = processor()
    disk = disk()
    ram = ram()
    totalPrice = 0

    def __init__(self, processor = processor, disk = disk, ram = ram, totalPrice = 0):
        self.__processor = processor
        self.__disk = disk
        self.__ram = ram
        self.__totalPrice = processor.getPrice() + disk.getPrice() + ram.getPrice()

    #prosedur setter
    def setProcessor(self, processor):
        self.__processor = processor
    
    def setDisk(self, disk):
        self.__disk = disk
    
    def setRam(self, ram):
        self.__ram = ram
    
    def setTotalPrice(self, ProcPrice, DiskPrice, RamPrice):
        self.__totalPrice = ProcPrice + DiskPrice + RamPrice

    #prosedur getter data
    def getProcessor(self):
        return self.__processor
    
    def getDisk(self):
        return self.__disk
    
    def getRam(self):
        return self.__ram
    
    def getTotalPrice(self):
        return self.__totalPrice

File: java/Pc.java
public class Pc {
    
    private Processor processor;
    private Disk disk;
    private Ram ram;
    private int totalPrice;

    //konstruktor
    Pc(){ }

    Pc(Processor processor, Disk disk, Ram ram, int totalPrice){
        this.processor = processor;
        this.disk = disk;
        this.ram = ram;
        this.totalPrice = processor.getPrice() + disk.getPrice() + ram.getPrice();
    }

    //prosedur setter
    public void setProcessor(Processor processor){
        this.processor = processor;
    }

    public void setDisk(Disk disk){
        this.disk = disk;
    }

    public void setRam(Ram ram){
        this.ram = ram;
    }

    public void totalPrice(int ProcPrice, int DiskPrice, int RamPrice){
        this.totalPrice = ProcPrice + DiskPrice + RamPrice;
    }

    //prosedur getter
    public Processor getProcessor(){
        return this.processor;
    }

    public Disk getDisk(){
        return this.disk;
    }

    public Ram getRam(){
        return this.ram;
    }

    public int getTotalPrice(){
        return this.totalPrice;
    }

}
